docs/source/dev-setup/devenv.rst

```diff
@@ -46,6 +46,125 @@ Prerequisites
 
     brew install jq
 
+Go Programming Language
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Hyperledger Fabric uses the Go Programming Language for many of its
+components.
+
+  - `Go <https://golang.org/dl/>`__ version 1.14.x is required.
+
+There are two environment variables you will need to set properly; you can make these settings permanent by placing them in the appropriate startup file, such as your personal ``~/.bashrc`` file if you are using the ``bash`` shell under Linux.
+
+First, you must set the environment variable ``GOPATH`` to point at the
+Go workspace containing the downloaded Fabric code base, with something like:
+
+.. code:: bash
+  export GOPATH=$HOME/go
+.. note:: You **must** set the GOPATH variable
+
+  Even though, in Linux, Go's ``GOPATH`` variable can be a colon-separated list
+  of directories, and will use a default value of ``$HOME/go`` if it is unset,
+  the current Fabric build framework still requires you to set and export that
+  variable, and it must contain **only** the single directory name for your Go
+  workspace. (This restriction might be removed in a future release.)
+
+Second, you should (again, in the appropriate startup file) extend your
+command search path to include the Go ``bin`` directory, such as the following
+example for ``bash`` under Linux:
+
+.. code:: bash
+  export PATH=$PATH:$GOPATH/bin
+While this directory may not exist in a new Go workspace installation, it is
+populated later by the Fabric build system with a small number of Go executables
+used by other parts of the build system. So even if you currently have no such
+directory yet, extend your shell search path as above.
+
+Developing on Windows
+~~~~~~~~~~~~~~~~~~~~~
+
+If you are developing on Windows 7, you will want to work within the
```

Comment:
Someone else should really review this. With WSL and WSL2, I really hope the whole MSYS thing can go.

I'd really like to see this whole thing go away in favor of a simple statement: We don't build or test on Windows.

Reply:
Agreed, the number of threads that get started in RocketChat only to later discover they were running on native Windows is a pain. A simple statement as Matt has pointed out would at least help drive home the point (it being in the official doc) rather than just us arguing in PR's and RC that its not supported.

I'm patiently waiting for WSL2 to come out of Insider Preview to out the experience.

Reply:
I think it is reasonable to drop Windows 7 at this point given that it's no longer supported and companies are forced to give up on them whether they like it or not. Before dropping support for Windows altogether though I would suggest we look at how many people use Windows for Fabric. I remember last time @christo4ferris checked he was astonished to see that the majority of our users were actually on Windows. This has probably changed but it'd be good to know. The larger industry doesn't move as quickly as we tend to believe from within our bubble. We could also ask the fabric mailing list to see what people say.
If we could get away with merely stating that people should be able to use WSL or WSL2 that'd be nice but it'd be best if someone could try it out and confirm (I don't have a Windows 10 machine, @mastersingh24 does :-). That's actually something we could ask the community to help with if they care to see support continue.

Reply:
I've removed all references to Windows 7 now from the docs.
We still need to agree about what to do about Windows 10.

Reply:
This section is about the prerequisites for developing the core Fabric code base and not about users developing applications or chaincode for Fabric.

> Before dropping support for Windows altogether

Since I started working with Fabric in 2018, we have never had any build, continuous integration, or automated testing on Windows; this isn't "dropping" support as much as it's clearly stating the current state of things.

In terms of "support", we support windows for the client side. That includes things like packaging chaincode, invoking transactions, and generating crypto material. This does not include running the peer node or orderer.

> majority of our users

This is not the same as the "majority of developers." If people want to develop Fabric in a Windows environment, they're swimming upstream and should not expect a smooth experience. We should not sugar coat that.

> we could ask the community to help with if they care to see support continue

Again, it's not "continue" as much as "start." If Windows is a required platform for Fabric development and node/orderer execution, I would expect a proposal and a focused effort around it as it would require additional tooling and documentation; I've said as much on the most recent PRs related to issues encountered when running on Windows.